src/util-decode-asn1.c

```diff
@@ -146,15 +146,15 @@ static uint32_t SCAsn1GetLengthLongForm(Asn1Ctx *ac)
             return ASN1_PARSER_ERR;
         }
 
-        if ((uint64_t) ((uint64_t)content_len +
-            (uint64_t) ASN1_BER_GET_HIGH_TAG_NUM(raw_len)) > UINT32_MAX)
+        uint64_t tmp_len = ((uint64_t)content_len << 8) + (uint64_t) raw_len;
+        if (tmp_len > UINT32_MAX)
         {
             node->flags |= ASN1_BER_EVENT_LEN_TOO_LONG;
             ac->parser_status = ASN1_STATUS_INVALID;
             return ASN1_PARSER_ERR;
         }
 
-        content_len += raw_len;
+        content_len = tmp_len;
```

Comment:
Do we still need this to be in the loop `for (; i < oct_len; i++)` ?

Reply:
What about ASN1 `8400010000` ?
Should not it be 4 bytes length encoded for 0x10000 so 65536 ?

Reply:
Hey @catenacyber, we still need the for loop in order to gather the `oct_len` bytes into `content_len`, your example is correct.